pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -52,24 +56,25 @@ export async function readDeploymentProfiles(): Promise<settings.DeploymentProfi
 
   switch (os.platform()) {
   case 'win32':
-    for (const key of [nativeReg.HKLM, nativeReg.HKCU]) {
+    for (const keyName of ['HKLM', 'HKCU'] as const) {
+      const key = nativeReg[keyName];
       const registryKey = nativeReg.openKey(key, REGISTRY_PATH_PROFILE.join('\\'), nativeReg.Access.READ);
 
       if (!registryKey) {
         continue;
       }
-      const defaultsKey = nativeReg.openKey(registryKey, 'Defaults', nativeReg.Access.READ);
-      const lockedKey = nativeReg.openKey(registryKey, 'Locked', nativeReg.Access.READ);
+      const defaultsKey = nativeReg.openKey(registryKey, defaultsHiveName, nativeReg.Access.READ);
+      const lockedKey = nativeReg.openKey(registryKey, lockedHiveName, nativeReg.Access.READ);
 
       try {
         if (defaultsKey) {
-          defaults = readRegistryUsingSchema(settings.defaultSettings, defaultsKey) ?? {};
+          defaults = readRegistryUsingSchema(settings.defaultSettings, defaultsKey, [keyName, defaultsHiveName]) ?? {};
```

Comment:
I've split into two parts -- we only need the info in `REGISTRY_PATH_PROFILE` for error messages, because all accesses are done via the current registry key (which acts like a database cursor, but through a tree instead of a table of rows).